### PR TITLE
extensions/kde-neon: use platform snap as plug name

### PR DESCRIPTION
**Pull request by ppd**
Using the same plug name "kde-frameworks-5-plug"
for all providers leads to problems when an
application updates its base or changes its default
provider because a new framework snap gets published
for the same base.

In that case, snapd will keep the old framework snap
connected and will happily connect the new framework
snap in addition to the old one.
This breaks applications because they won't find the correct
framework snap mounted at $SNAP/kf5.

Prevent this by using the same strategy that the gnome
extensions use: plug name = content snap name

See https://forum.snapcraft.io/t/snap-accumulates-content-interface-connections-to-multiple-platform-snaps/27523/8

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh tests/unit/project_loader/extensions`?

-----

I built a snap (anki-ppd) with this change and everything works as expected. Also CC @jriddell.
I'm not sure about the subsystem prefix; it seemed more complete than just "extensions".

### CODE REVIEW

File: snapcraft/internal/project_loader/_extensions/kde_neon.py [hunk 1]
```diff
@@ -126,7 +126,7 @@ def __init__(self, *, extension_name: str, yaml_data: Dict[str, Any]) -> None:
                 "source": "$SNAPCRAFT_EXTENSIONS_DIR/desktop",
                 "source-subdir": "kde-neon",
                 "plugin": "make",
-                "make-parameters": ["PLATFORM_PLUG=kde-frameworks-5-plug"],
+                "make-parameters": ["PLATFORM_PLUG={plug}".format(plug=info.provider)],
```
Comment (sergiusens) [inline snapcraft/internal/project_loader/_extensions/kde_neon.py:129]:
```suggestion
                "make-parameters": [f"PLATFORM_PLUG={info.provider}"],
```

**Review by sergiusens — COMMENTED** [review]
Looks good, thanks.

File: snapcraft/internal/project_loader/_extensions/kde_neon.py (hunk 1 above)
Comment (ppd) [reply snapcraft/internal/project_loader/_extensions/kde_neon.py]: Changed, thanks

**Comment by ppd** [comment]
I also removed the manual connection in the spread test. It is a) not required any more because auto-connect has been granted and b) the plug name is different now, so we'd have to change it anyway.

**Review by sergiusens — APPROVED** [review]

**Review by sergiusens — APPROVED** [review]
thank you!

**Pull request merged**